lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2228,8 +2228,55 @@ namespace Js
         }
 
         ScriptContext* scriptContext = pThis->type->GetScriptContext();
-
         ApiError error = ApiError::NoError;
+        const char16 *pThisSz = pThis->GetSz();
+        charcount_t pThisLength = pThis->GetLength();
+
+        bool isAscii = true;
+        for (charcount_t i = 0; i < pThisLength; i++)
+        {
+            if (pThisSz[i] >= 0x80)
+            {
+                isAscii = false;
+                break;
+            }
+        }
+
+        if (isAscii)
```

Comment:
I think all of this ascii stuff needs to be inside a `if (useInvariant)` block; otherwise `"i".toLocaleUpperCase("tr-tr")` would be wrong.